src/editor/components/scenegraph/AIChatPanel.js

```diff
@@ -0,0 +1,374 @@
+            }),
+            component: Schema.string({
+              description:
+                'The component to update (e.g., position, rotation, mixin)'
+            }),
+            property: Schema.string({
+              description:
+                'The property to update within the component (optional)'
+            }),
+            value: Schema.string({
+              description: 'The new value to set'
+            }),
+            'expression-for-value': Schema.string({
+              description:
+                'Mathematical expression to evaluate for the value (e.g., "5 - 2"). Use this instead of value when calculation is needed.'
+            })
+          },
+          optionalProperties: ['value', 'expression-for-value', 'property']
+        })
+      }
+    ]
+  };
+
+  useEffect(() => {
+    console.log('AIChatPanel mounted');
+    const initializeAI = async () => {
+      try {
+        console.log('Initializing Vertex AI');
+        const model = getGenerativeModel(vertexAI, {
+          model: 'gemini-2.0-flash',
+          tools: entityTools
+        });
+        // Initialize chat with history
+        modelRef.current = model.startChat({
+          history: [],
+          generationConfig: {
+            maxOutputTokens: 1000
+          }
+        });
+        console.log('Vertex AI chat initialized successfully');
+      } catch (error) {
+        console.error('Error initializing Vertex AI:', error);
+      }
+    };
+
+    initializeAI();
+  }, []);
+
+  const handleSendMessage = async () => {
+    if (!input.trim() || !modelRef.current) return;
+
+    setIsLoading(true);
+    const userMessage = { role: 'user', content: input };
+    setMessages((prev) => [...prev, userMessage]);
+    setInput('');
+    let aiMessage;
+
+    try {
+      const entity = document.getElementById('street-container');
+      const data = STREET.utils.convertDOMElToObject(entity);
+      const filteredData = STREET.utils.filterJSONstreet(data);
+      const sceneJSON = JSON.parse(filteredData).data;
+
+      const prompt = `
+      Context: You are a 3D street scene assistant for the 3DStreet application. 
+      The current scene has the following state:
+      ${JSON.stringify(sceneJSON, null, 2)}
+
+      User request: ${input}
+
+      Please analyze the request and provide one of the following:
+      1. If the user is asking about the scene, provide a natural language explanation
+      2. If the user is asking to modify the scene, use the entityUpdate function
+      3. If the user needs help, provide relevant guidance about the 3DStreet editor
+
+      In the scene state, units for length are in meters, and rotations are in degrees.
+
+      The orientation of axis to cardinal directions is as follows: x+ (positive) is north; x- (negative) is south; y+ (positive) is up; y- (negative) is down; z- (negative) is west; z+ (positive) is east;
+      Models face z- (east) when at 0º Y rotation. 
```

Comment:
z+

Reply:
It's better not to confuse the AI :D

Reply:
aca44d9ac6f302863352e6be5fd1b78b86d11337